server/src/main/java/org/opensearch/wlm/cancellation/AbstractTaskSelectionStrategy.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.wlm.cancellation;
+
+import org.opensearch.cluster.metadata.QueryGroup;
+import org.opensearch.search.ResourceType;
+import org.opensearch.tasks.CancellableTask;
+import org.opensearch.tasks.Task;
+import org.opensearch.tasks.TaskCancellation;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Represents an abstract task selection strategy.
+ * This class implements the TaskSelectionStrategy interface and provides a method to select tasks for cancellation based on a sorting condition.
+ * The specific sorting condition depends on the implementation.
+ */
+public abstract class AbstractTaskSelectionStrategy implements TaskSelectionStrategy {
```

Comment:
Do we really need the interface if abstract class is implementing it? Can we not declare the below method as abstract in this class?

```
public abstract List<TaskCancellation> selectTasksForCancellation(QueryGroup queryGroup, List<Task> tasks, long limit, ResourceType resourceType);
```

Reply:
ya i think that's a valid feedback, i can get rid of the interface and include the implementation of `selectTasksForCancellation` in `AbstractTaskSelectionStrategy `, though i'll keep it non final and let the concrete classes of `AbstractTaskSelectionStrategy` to write its own implementation if necessary.